File: apps/backend-java/src/main/java/com/enterpriseagentos/backendjava/domain/features/agents/AgentRunRecord.java
package com.enterpriseagentos.backendjava.domain.features.agents;

import java.math.BigDecimal;
import java.time.Instant;
import java.time.YearMonth;
import java.util.*;
import java.util.concurrent.CompletableFuture;
import com.enterpriseagentos.backendjava.domain.common.primitives.*;
import com.enterpriseagentos.backendjava.domain.common.services.*;
import com.enterpriseagentos.backendjava.domain.common.valueobjects.*;
import com.enterpriseagentos.backendjava.domain.events.*;
import com.enterpriseagentos.backendjava.domain.features.agents.*;
import com.enterpriseagentos.backendjava.domain.features.analytics.*;
import com.enterpriseagentos.backendjava.domain.features.management.*;
import com.enterpriseagentos.backendjava.domain.features.mcp.*;

public class AgentRunRecord {
    public UUID id;
    public UUID agentId;
    public UUID parentRunId;
    public String parentCorrelationId;
    public String kind;
    public String status;
    public String name;
    public String description;
    public String prompt;
    public String result;
    public String error;
    public Instant createdAt;
    public Instant updatedAt;
    public Instant completedAt;
}

File: apps/backend/src/Features/Agents/Domain/AgentRunRecord.cs
namespace OffceOs.Domain.Features.Agents;

public sealed class AgentRunRecord
{
    public Guid Id { get; init; } = Guid.NewGuid();
    public Guid AgentId { get; init; }
    public Guid? ParentRunId { get; init; }
    public string? ParentCorrelationId { get; init; }
    public string Kind { get; init; } = "turn";
    public string Status { get; set; } = "running";
    public string Name { get; init; } = string.Empty;
    public string? Description { get; init; }
    public string Prompt { get; init; } = string.Empty;
    public string? Result { get; set; }
    public string? Error { get; set; }
    public DateTime CreatedAt { get; init; } = DateTime.UtcNow;
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
    public DateTime? CompletedAt { get; set; }
}
